### ISSUE TITLE
[a11y]: Component with "menu" role does not have a tabbable element

### ISSUE THREAD

**Issue by vasifrmn** [carbon-design-system/carbon#19551]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.81.0

### React version

18.2.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

Menu component with role="menu" lacks a tabbable element, causing accessibility violations

![Image](https://github.com/user-attachments/assets/2130e272-c1b3-4222-bc85-7ca5caa3a711)

### Reproduction/example

https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-19-vite?file=src%2FApp.jsx&preset=node=

### Steps to reproduce

All the Menu and MenuItem components have issues.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by heloiselui** [comment]
Thanks for open this issue @vasifrmn.

The Menu component at https://react.carbondesignsystem.com/?path=/story/components-menu--default fails accessibility checks. This breaks expected keyboard interaction per Carbon’s guidelines: https://carbondesignsystem.com/components/menu/accessibility/. 

https://github.com/user-attachments/assets/dabe61ec-0451-4453-8eed-f195b51b9c32
